.azure/pipelines/jobs/default-build.yml

```diff
@@ -157,6 +156,7 @@ jobs:
     - LC_ALL: 'en_US.UTF-8'
     - LANG: 'en_US.UTF-8'
     - LANGUAGE: 'en_US.UTF-8'
+    - skipComponentGovernanceDetection: ${{ ne(variables['System.TeamProject'], 'internal') }}
```

Comment:
I'm not sure this will work. Might need something more like https://github.com/dotnet/aspnetcore/blob/4b60b307b6e7550816ced673a32fcf5f8c7b1be2/.azure/pipelines/jobs/default-build.yml#L271-L275 because CG runs as an injected build step. 

**If** it works, it's nicer than what we have down there and I prefer this approach. But please try deleting those lines as well as https://github.com/dotnet/aspnetcore/blob/4b60b307b6e7550816ced673a32fcf5f8c7b1be2/.azure/pipelines/jobs/default-build.yml#L36-L37 and https://github.com/dotnet/aspnetcore/blob/4b60b307b6e7550816ced673a32fcf5f8c7b1be2/.azure/pipelines/jobs/default-build.yml#L102-L103

Reply:
/fyi @mmitche I bet the following is useless in Arcade these days: https://github.com/dotnet/arcade/blob/main/eng/common/templates/job/job.yml#L145-L147

If the task weren't auto-injected, we'd never see it in public rolling builds.

Reply:
I believe this was added because it only auto-injects on main?

Reply:
Trying a public build of this branch first to see if it still runs CG: https://dev.azure.com/dnceng/public/_build/results?buildId=1961562&view=results

Reply:
It skipped it: https://dev.azure.com/dnceng/public/_build/results?buildId=1961562&view=logs&j=3f6d4e0f-1b71-56b5-361e-d95b6e6da15a. I'll try again after removing the unnecessary stuff you mentioned

Reply:
New build: https://dev.azure.com/dnceng/public/_build/results?buildId=1961822&view=results

Reply:
Still works - @dougbu any other concerns?

Reply:
nope